src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -127,10 +127,24 @@ fun inferExpressionType(expr: RsExpr): RustType {
                 RustUnknownType
         }
 
+        is RsArrayExpr -> inferArrayType(expr)
+
         else -> RustUnknownType
     }
 }
 
+private fun inferArrayType(expr: RsArrayExpr): RustType {
+    val (elementType, size) = if (expr.semicolon != null) {
+        if (expr.exprList.size != 2) return RustUnknownType
```

Comment:
It's better to return `RsArrayType(RustUnknownType)` rather than `RustUnknownType`. 

In general in the IDE the goal is to infer as may types as possible, and not to find a type error as early as possible.